### PR TITLE
Update resource manager readme

### PR DESCRIPTION
**Pull request by MarcoPolo**
Moves https://github.com/libp2p/go-libp2p-resource-manager/pull/78/files to here.

### CODE REVIEW

File: p2p/host/resource-manager/limit_defaults.go [hunk 1]
```diff
@@ -255,54 +255,50 @@ func (cfg *LimitConfig) Apply(c LimitConfig) {
 
 // Scale scales up a limit configuration.
 // memory is the amount of memory that the stack is allowed to consume,
-// for a full it's recommended to use 1/8 of the installed system memory.
+// for a dedicated node it's recommended to use 1/8 of the installed system memory.
 // If memory is smaller than 128 MB, the base configuration will be used.
 func (cfg *ScalingLimitConfig) Scale(memory int64, numFD int) LimitConfig {
-	var scaleFactor int
```
Comment (marten-seemann) [inline p2p/host/resource-manager/limit_defaults.go:261]: Can we move that code change to a separate PR? Otherwise this will be very confusing if we ever have to go back in the commit history.

**Review by marten-seemann — CHANGES_REQUESTED** [review]

**Review by BigLep — APPROVED** [review]
Providing approval given https://github.com/libp2p/go-libp2p-resource-manager/pull/78 , but understood there is a request to break out some of the commits.

File: p2p/host/resource-manager/limit_defaults.go (hunk 1 above)
Comment (MarcoPolo) [reply p2p/host/resource-manager/limit_defaults.go]: Done https://github.com/libp2p/go-libp2p/pull/1685
